cmd/thor/node/packer_loop.go

```diff
@@ -106,107 +110,186 @@ func (n *Node) packerLoop(ctx context.Context) {
+
+	if err := n.writeLogsIfEnabled(ctx, newBlock, receipts); err != nil {
+		return errors.Wrap(err, "write logs")
+	}
+
+	if err := n.commitState(stage); err != nil {
+		return errors.Wrap(err, "commit state")
+	}
+
+	if err := n.syncLogWorker(ctx); err != nil {
+		return err
+	}
+
+	if err := n.addBlockToRepository(newBlock, receipts, ctx.conflicts); err != nil {
+		return errors.Wrap(err, "add block")
+	}
 
-		n.processFork(newBlock, oldBest.Header.ID())
-		commitElapsed := mclock.Now() - startTime - execElapsed
+	if err := n.commitToBFT(newBlock); err != nil {
+		return errors.Wrap(err, "bft commits")
+	}
+
+	return n.finalizeAndBroadcast(ctx, newBlock, receipts, execElapsed)
+}
+
+func (n *Node) writeLogsIfEnabled(ctx *packContext, newBlock *block.Block, receipts tx.Receipts) error {
+	if ctx.logEnabled {
+		return n.writeLogs(newBlock, receipts, ctx.oldBest.Header.ID())
+	}
+	return nil
+}
 
-		n.comm.BroadcastBlock(newBlock)
-		logger.Info("📦 new block packed",
-			"txs", len(receipts),
-			"mgas", float64(newBlock.Header().GasUsed())/1000/1000,
-			"et", fmt.Sprintf("%v|%v", common.PrettyDuration(execElapsed), common.PrettyDuration(commitElapsed)),
-			"id", shortID(newBlock.Header().ID()),
-		)
+func (n *Node) commitState(stage *state.Stage) error {
+	_, err := stage.Commit()
+	return err
+}
 
-		if v, updated := n.bandwidth.Update(newBlock.Header(), time.Duration(realElapsed)); updated {
-			logger.Trace("bandwidth updated", "gps", v)
+func (n *Node) syncLogWorker(ctx *packContext) error {
+	if ctx.logEnabled {
+		if err := n.logWorker.Sync(); err != nil {
+			log.Warn("failed to write logs", "err", err)
+			n.logDBFailed = true
 		}
+	}
+	return nil
+}
 
-		metricBlockProcessedTxs().SetWithLabel(int64(len(receipts)), map[string]string{"type": "proposed"})
-		metricBlockProcessedGas().SetWithLabel(int64(newBlock.Header().GasUsed()), map[string]string{"type": "proposed"})
-		metricBlockProcessedDuration().Observe(time.Duration(realElapsed).Milliseconds())
-		return nil
-	})
+func (n *Node) addBlockToRepository(newBlock *block.Block, receipts tx.Receipts, conflicts uint32) error {
+	return n.repo.AddBlock(newBlock, receipts, conflicts, true)
+}
+
+func (n *Node) commitToBFT(newBlock *block.Block) error {
+	if newBlock.Header().Number() >= n.forkConfig.FINALITY {
+		return n.bft.CommitBlock(newBlock.Header(), true)
+	}
+	return nil
+}
+
+func (n *Node) finalizeAndBroadcast(ctx *packContext, newBlock *block.Block, receipts tx.Receipts, execElapsed mclock.AbsTime) error {
+	realElapsed := mclock.Now() - ctx.startTime
+	commitElapsed := realElapsed - execElapsed
+
+	n.processFork(newBlock, ctx.oldBest.Header.ID())
+	n.comm.BroadcastBlock(newBlock)
+
+	n.logBlockPacked(newBlock, receipts, execElapsed, commitElapsed)
+	n.updateMetrics(newBlock, receipts, realElapsed)
+
+	return nil
+}
+
+func (n *Node) logBlockPacked(newBlock *block.Block, receipts tx.Receipts, execElapsed, commitElapsed mclock.AbsTime) {
+	logger.Info("�� new block packed",
```

Comment:
Did the box emoji broke?
```suggestion
	logger.Info("📦 new block packed",
```